generator-engine/src/main/java/hu/blackbelt/judo/meta/psm/generator/engine/PsmGenerator.java

```diff
@@ -333,4 +343,49 @@ public static void recalculateChecksumForDirectory(PsmGeneratorParameter paramet
         ModelGenerator.recalculateChecksumToDirectory(genericParams, applications);
     }
 
+    public static String generalizeName(String str) {
+        if (str == null || str.length() == 0) {
+            return str;
+        }
+        boolean isUpperSnakeCae = !str.matches(negateRegex("([A-Z_0-9]*)"));
+        if (isUpperSnakeCae) {
+            return toCamelCase(Arrays.stream(str.split("_")).map(s -> capitalize(s)).collect(Collectors.joining()));
+        } else if (str.matches(".*[.#@_,;:-].*")) {
+            return toCamelCase(Arrays.stream(str.split("[.#@_,;:-]")).map(s -> firstToUpper(s)).collect(Collectors.joining()));
+        }
+        return toCamelCase(str);
+    }
+
+    public static  String negateRegex(String regex) {
+        return "(?!(?:" + regex + ")$).*";
+    }
+
+    public static String toCamelCase(String str) {
+        String[] words = str.split("(?=[A-Z])");
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < words.length; i++) {
+            String word = words[i];
+            if (i == 0) {
+                word = word.isEmpty() ? word : word.toLowerCase();
+            } else {
+                word = word.isEmpty() ? word : capitalize(word);
+            }
+            builder.append(word);
+        }
+        return builder.toString();
+    }
+    public static String firstToUpper(String str) {
+        if (str != null && str.length() > 1) {
+            return str.substring(0, 1).toUpperCase() + str.substring(1);
+        }
+        return str;
+    }
+
+    private static String capitalize(String str) {
+        if (str != null && str.length() > 1) {
+            return str.substring(0,1).toUpperCase() + str.substring(1).toLowerCase();
+        }
+        return str;
+    }
```

Comment:
Capitalize should first uppercase every "word" after a whitespace, am i right?

Reply:
No :)